src/main/java/frc/robot/hardware/Phoenix6Device.java

```diff
@@ -0,0 +1,41 @@
+package frc.robot.hardware;
+
+import com.ctre.phoenix6.BaseStatusSignal;
+import com.ctre.phoenix6.StatusSignal;
+import frc.robot.hardware.signal.InputSignal;
+import frc.robot.hardware.signal.phoenix.Phoenix6BothLatencySignal;
+import frc.robot.hardware.signal.phoenix.Phoenix6SignalBuilder;
+import org.littletonrobotics.junction.Logger;
+
+import java.util.LinkedList;
+
+public abstract class Phoenix6Device implements IDevice {
+
+	private final ConnectedInputAutoLogged connectedInput;
+	private final String logPath;
+
+	public Phoenix6Device(String logPath) {
+		this.connectedInput = new ConnectedInputAutoLogged();
+		this.logPath = logPath;
+	}
+
+	public boolean isConnected() {
+		return connectedInput.connected;
+	}
+
+	public void updateSignals(InputSignal... signals) {
+		LinkedList<StatusSignal<Double>> signalsSet = new LinkedList<>();
+		for (InputSignal signal : signals) {
+			if (signal instanceof Phoenix6SignalBuilder.SignalGetter) {
+				signalsSet.add(((Phoenix6SignalBuilder.SignalGetter) signal).getSignal());
+				if (signal instanceof Phoenix6BothLatencySignal) {
+					signalsSet.add(((Phoenix6BothLatencySignal) signal).getSignalSlope());
+				}
+			}
+		}
+
+		connectedInput.connected = BaseStatusSignal.refreshAll(signalsSet.toArray(StatusSignal[]::new)).isOK();
+		Logger.processInputs(logPath, connectedInput);
```

Comment:
need to add a for loop to referesh each or it wont update the value:
```suggestion
		for (InputSignal signal : signals){
		    Logger.processInputs(logpath, signal)
		}
```

Reply:
btw its noam

Reply:
i dont want to process here, like i can but I thought it will be like now when the proccess happen in subsystem

Reply:
I understand where you come from, but there are 2 things I think needs to be taken into account:
- if it's processed here, there will not be any mistakes or "oh I forgot" in the subsystem layer
- secondly, if you are getting a signal from the motor, why not put it in the log? it's more information you are already getting.